projects/packages/autoloader/tests/php/tests/unit/ShutdownHandlerTest.php

```diff
@@ -0,0 +1,101 @@
+<?php // phpcs:ignore WordPress.Files.FileName
+/**
+ * Shutdown handler test suite.
+ *
+ * @package automattic/jetpack-autoloader
+ */
+
+// We live in the namespace of the test autoloader to avoid many use statements.
+namespace Automattic\Jetpack\Autoloader\jpCurrent;
+
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Test suite class for the Shutdown_Handler.
+ */
+class ShutdownHandlerTest extends TestCase {
+
+	/**
+	 * The mock Plugins_Handler instance.
+	 *
+	 * @var Plugins_Handler|\PHPUnit\Framework\MockObject\MockObject
+	 */
+	private $plugins_handler;
+
+	/**
+	 * Setup runs before each test.
+	 *
+	 * @before
+	 */
+	public function set_up() {
+		$this->plugins_handler = $this->getMockBuilder( Plugins_Handler::class )
+			->disableOriginalConstructor()
+			->getMock();
+	}
+
+	/**
+	 * Tests that the shutdown handler caches the active plugins.
+	 */
+	public function test_shutdown_caches_active_plugins() {
+		$this->plugins_handler->expects( $this->once() )
+			->method( 'get_active_plugins' )
+			->with( false, true )
+			->willReturn( array( TEST_PLUGIN_DIR ) );
+		$this->plugins_handler->expects( $this->once() )
+			->method( 'cache_plugins' )
+			->with( array( TEST_PLUGIN_DIR ) );
+
+		// Mark that the plugins have been loaded so that we can perform a safe shutdown.
+		do_action( 'plugins_loaded' );
+
+		( new Shutdown_Handler( $this->plugins_handler, array(), false ) )();
```

Comment:
Looks like you'll need to solve this so the tests don't fail.

Reply:
It looks like that syntax isn't compatible with PHP 5.6. We can probably just assign the object to a variable first then invoke it.

Reply:
I've fixed the test.